pkg/storer/internal/pinning/pinning.go

```diff
@@ -266,12 +266,12 @@ func (c *collectionPutter) Close(st internal.Storage, root swarm.Address) error
 
 	// Save the root pin reference.
 	c.collection.Addr = root
-	err := st.IndexStore().Put(c.collection)
+	err := writer.Put(c.collection)
 	if err != nil {
 		return fmt.Errorf("pin store: failed updating collection: %w", err)
 	}
 
-	err = st.IndexStore().Delete(&dirtyCollection{UUID: c.collection.UUID})
+	err = writer.Delete(&dirtyCollection{UUID: c.collection.UUID})
```

Comment:
In the pinstore, the deleteCollectionChunks method could benefit a lot from use of batch. We missed that.